### PR TITLE
Remove CriteriaBuilder and use repository methods instead

### PR DESCRIPTION
**Pull request by jjbuchan**
# What

Removes all the `CriteriaBuilder` logic for building sql queries and replaces it by using standard repository options.

# How

The "how" is fairly obvious.  The only real decision was to start using `Optional` fields where we are getting at most one returned object.  I like this overall, but the downside is it makes the tests a little more tedious due to having to do various `isPresent()` and `get()` calls.

## How to test

Run all tests.  They still work!

# Why

Cleans up the code and standardizes things across all repos.

# TODO

Incoming PRs for model repo... I think that'll be it.

### CODE REVIEW

File: src/main/java/com/rackspace/salus/monitor_management/repositories/MonitorRepository.java [hunk 1]
```diff
@@ -17,10 +17,18 @@
 package com.rackspace.salus.monitor_management.repositories;
 
 import com.rackspace.salus.telemetry.model.Monitor;
+
+import java.util.Optional;
 import java.util.UUID;
+import java.util.stream.Stream;
```
Comment (harageth) [inline src/main/java/com/rackspace/salus/monitor_management/repositories/MonitorRepository.java]: I dont think this or the java.util.Optional are being used here anymore

File: src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java [hunk 2]
```diff
@@ -172,14 +146,9 @@ public Monitor getMonitor(String tenantId, UUID id) {
      * @return Stream of monitors.
      */
     public Stream<Monitor> getMonitorsAsStream() {
-        CriteriaBuilder cb = entityManager.getCriteriaBuilder();
-        CriteriaQuery<Monitor> cr = cb.createQuery(Monitor.class);
-        Root<Monitor> root = cr.from(Monitor.class);
-        cr.select(root);
-        return entityManager.createQuery(cr).getResultStream();
+        return Streams.stream(monitorRepository.findAll());
```
Comment (harageth) [inline src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java]: Considering the other stuff just wasnt working this looks a lot easier.

File: src/main/java/com/rackspace/salus/monitor_management/repositories/MonitorRepository.java (hunk 1 above)
Comment (jjbuchan) [reply src/main/java/com/rackspace/salus/monitor_management/repositories/MonitorRepository.java]: Correct.  Removed them now.

File: src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java (hunk 2 above)
Comment (itzg) [inline src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java]: Yeah, it's strange how going from an Iterable to a Stream is awkward even though they seem close in concept. I guess the non-Guava might be

```
StreamSupport.stream(
  monitorRepository.findAll().spliterator(),
  false
)
```

Comment (jjbuchan) [reply src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java]: Should we be doing it that way instead?  I know no reason to not use guava other than it saying it's unstable... but you've said before that it doesn't matter.

I'm happy to go either way.

Comment (itzg) [reply src/main/java/com/rackspace/salus/monitor_management/services/MonitorManagement.java]: I like guava, so I'd say keep it as you have it. What I found is just something we can try next time that's the reason we need to pull in guava.

**Pull request merged**